src/include/txn_inline.h

```diff
@@ -115,7 +115,7 @@ __wt_txn_op_set_key(WT_SESSION_IMPL *session, const WT_ITEM *key)
     op = txn->mod + txn->mod_count - 1;
 
     if (WT_SESSION_IS_CHECKPOINT(session) || WT_IS_HS(op->btree->dhandle) ||
```

Comment:
Can we add some text either here or in the function comment that this behaviour is only required when the transaction might be prepared?